File: src/main/java/com/ai2dev/devninja_ai2dev_lambda/exception/ApiException.java
package com.ai2dev.devninja_ai2dev_lambda.exception;

public class ApiException extends RuntimeException {
    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(ApiException.class);

    private final int statusCode;

    public ApiException(int statusCode, String message) {
        super(message);
        LOG.info("{\"event\":\"entry\"}");
        this.statusCode = statusCode;
        LOG.info("{\"event\":\"exit\",\"id\":\"{}\"}", statusCode);
    }

    public int getStatusCode() {
        return statusCode;
    }
}

File: Helpers/ApiException.cs
using System;
using System.Net;

namespace AdverseEventReporter.Helpers;

public class ApiException : Exception
{
    public HttpStatusCode StatusCode { get; }
    public string Code { get; }

    public ApiException(HttpStatusCode statusCode, string code, string message) : base(message)
    {
        StatusCode = statusCode;
        Code = code;
    }
}
